src/modules/ipfs/pin.processor.ts

```diff
@@ -56,14 +54,29 @@ export class PinProcessor {
   async pin(job: Job<PinClaimData>) {
     const cid = job.data.cid;
     let claim = job.data.claim;
+    this.logger.debug(`Pinning CID ${cid}`);
     try {
       await this.didStoreCluster.get(cid);
     } catch (_) {
       if (!claim) {
-        claim = await this.didStoreInfura.get(cid);
+        try {
+          claim = await this.didStoreInfura.get(cid);
+        } catch (e) {
+          this.logger.error(
+            `Can not fetch claim from Infura IPFS. CID ${cid}: ${e}`
+          );
+          return;
+        }
+      }
+      try {
+        const clusterCid = await this.didStoreCluster.save(claim);
+        this.logger.debug(`CID ${cid} saved in cluster by CID ${clusterCid}`);
+      } catch (e) {
+        this.logger.error(
+          `Can not save claim in IPFS cluster. CID ${cid}: ${e}`
+        );
+        return;
```

Comment:
I think this change is okay to keep in this PR, but is it fair to say that it is somewhat separate from the main changes to the "incremental" did document syncing?

Reply:
That's fair. I moved this error handling in another PR https://github.com/energywebfoundation/ssi-hub/pull/676